drivers/adc/adc_nrfx_saadc.c

```diff
@@ -424,16 +454,16 @@ static const struct adc_driver_api adc_nrfx_driver_api = {
  * Just in case that assumption becomes invalid in the future, we use
  * a BUILD_ASSERT().
  */
-#define SAADC_INIT(inst)						\
-	BUILD_ASSERT((inst) == 0,					\
-		     "multiple instances not supported");		\
-	DEVICE_DT_INST_DEFINE(0,					\
-			    init_saadc,					\
-			    NULL,					\
-			    NULL,					\
-			    NULL,					\
-			    POST_KERNEL,				\
-			    CONFIG_ADC_INIT_PRIORITY,			\
-			    &adc_nrfx_driver_api);
+#define SAADC_INIT(inst)					\
+	BUILD_ASSERT((inst) == 0,				\
+		     "multiple instances not supported");	\
+	DEVICE_DT_INST_DEFINE(0,				\
+			      init_saadc,			\
+			      NULL,				\
+			      NULL,				\
+			      NULL,				\
+			      POST_KERNEL,			\
+			      CONFIG_ADC_INIT_PRIORITY,		\
+			      &adc_nrfx_driver_api);
 
 DT_INST_FOREACH_STATUS_OKAY(SAADC_INIT)
```

Comment:
nit: if multiple instances are not supported, then just drop `DT_INST_FOREACH_STATUS_OKAY`.